Modules/Filtering/ImageFeature/include/itkLaplacianSharpeningImageFilter.hxx

```diff
@@ -35,136 +33,53 @@ LaplacianSharpeningImageFilter<TInputImage, TOutputImage>::PrintSelf(std::ostrea
-  oper.CreateOperator();
-
-  // get a copy of the input requested region (should equal the output
-  // requested region)
-  typename TInputImage::RegionType inputRequestedRegion;
-  inputRequestedRegion = inputPtr->GetRequestedRegion();
-
-  // pad the input requested region by the operator radius
-  inputRequestedRegion.PadByRadius(oper.GetRadius());
-
-  // crop the input requested region at the input's largest possible region
-  if (inputRequestedRegion.Crop(inputPtr->GetLargestPossibleRegion()))
-  {
-    inputPtr->SetRequestedRegion(inputRequestedRegion);
-    return;
-  }
-  else
-  {
-    // Couldn't crop the region (requested region is outside the largest
-    // possible region).  Throw an exception.
-
-    // store what we tried to request (prior to trying to crop)
-    inputPtr->SetRequestedRegion(inputRequestedRegion);
-
-    // build an exception
-    InvalidRequestedRegionError e(__FILE__, __LINE__);
-    e.SetLocation(ITK_LOCATION);
-    e.SetDescription("Requested region is (at least partially) outside the largest possible region.");
-    e.SetDataObject(inputPtr);
-    throw e;
-  }
-}
-
 template <typename TInputImage, typename TOutputImage>
 void
 LaplacianSharpeningImageFilter<TInputImage, TOutputImage>::GenerateData()
 {
-  // Create the Laplacian operator
-  LaplacianOperator<RealType, ImageDimension> oper;
-  double                                      s[ImageDimension];
-  for (unsigned int i = 0; i < ImageDimension; ++i)
-  {
-    if (this->GetInput()->GetSpacing()[i] == 0.0)
-    {
-      itkExceptionMacro(<< "Image spacing cannot be zero");
-    }
-    else
-    {
-      s[i] = 1.0 / this->GetInput()->GetSpacing()[i];
-    }
-  }
-  oper.SetDerivativeScalings(s);
-  oper.CreateOperator();
 
-  // do calculations in floating point
-  using RealImageType = Image<RealType, ImageDimension>;
-  using NOIF = NeighborhoodOperatorImageFilter<InputImageType, RealImageType>;
-  ZeroFluxNeumannBoundaryCondition<InputImageType> nbc;
-
-  auto filter = NOIF::New();
-  filter->OverrideBoundaryCondition(static_cast<typename NOIF::ImageBoundaryConditionPointerType>(&nbc));
+  // Calculate the Laplacian filtered image
 
-  // Create a process accumulator for tracking the progress of this minipipeline
-  auto progress = ProgressAccumulator::New();
-  progress->SetMiniPipelineFilter(this);
-
-  // Register the filter with the with progress accumulator using
-  // equal weight proportion
-  progress->RegisterInternalFilter(filter, 0.8f);
+  using RealImageType = Image<RealType, ImageDimension>;
 
-  //
-  // set up the mini-pipeline
-  //
-  filter->SetOperator(oper);
-  filter->SetInput(this->GetInput());
-  filter->GetOutput()->SetRequestedRegion(this->GetOutput()->GetRequestedRegion());
+  using LaplacianImageFilter = LaplacianImageFilter<InputImageType, RealImageType>;
+  typename LaplacianImageFilter::Pointer laplacianFilter = LaplacianImageFilter::New();
+  laplacianFilter->SetInput(this->GetInput());
```

Comment:
The internal mini-pipeline needs to be isolated from the external. This is done with graft the input:
```
auto localInput = TInputImage::New();
localInput->Graft(this->GetInput());
laplacianFilter->SetInput(localInput);
```

Reply:
Thanks @blowekamp .  

Excuse the ignorance but I'm having a hard time following you.  First, is this assuming that this filter is capable of  streaming?  Because I assumed it wasn't and that all the "streaming" remnants were an artifact of copying over from the LaplacianImageFilter, which is capable of streaming.

Reply:
Ah, okay.  I'm a little slow this morning but I understand now (at least this part).  Thanks.